clusterloader2/pkg/measurement/common/api_availability_metrics.go

```diff
@@ -28,64 +28,51 @@ type apiAvailabilityMetrics struct {
 }
 
 type apiAvailabilitySummary struct {
-	IP                         string        `json:"IP"`
-	AvailabilityPercentage     float32       `json:"availabilityPercentage"`
-	LongestUnavailableDuration time.Duration `json:"longestUnavailableDuration"`
+	IP                       string  `json:"IP,omitempty"`
+	AvailabilityPercentage   float32 `json:"availabilityPercentage"`
+	LongestUnavailablePeriod string  `json:"longestUnavailablePeriod"`
 }
 
 type apiAvailabilityOutput struct {
-	ClusterMetrics *apiAvailabilitySummary   `json:"clusterMetrics"`
-	MasterMetrics  []*apiAvailabilitySummary `json:"masterMetrics"`
+	ClusterSummary apiAvailabilitySummary   `json:"clusterMetrics"`
+	MasterSummary  []apiAvailabilitySummary `json:"masterMetrics"`
+	Frequency      string                   `json:"frequency"`
 }
 
-func (output apiAvailabilityOutput) createClusterSummary(metrics *apiAvailabilityMetrics, probeFrequency int) {
-	metrics.buildAPIAvailabilityMetricsSummary(probeFrequency, true, "")
-}
-
-func (output apiAvailabilityOutput) createMastersSummary(metrics map[string]*apiAvailabilityMetrics, hosts []string, probeFrequency int) {
-	for _, host := range hosts {
-		masterHostAvailabilitySummary := metrics[host].buildAPIAvailabilityMetricsSummary(probeFrequency, false, host)
-		output.MasterMetrics = append(output.MasterMetrics, masterHostAvailabilitySummary)
+func (a *apiAvailabilityMetrics) update(availability bool) {
+	if availability {
+		a.numSuccesses++
```

Comment:
return

and everything else in the main flow of function - to reduce the nesting